Prepare roles for Ansible Galaxy

ndokos opened the pull request:
We have a reserved "pbench" namespace on Galaxy, so this is the"agent" collection under the "pbench" namespace. Galaxy uses "pbench.agent" most of the time, but there are cases where you have to tell it "pbench/agent" in order for it to understand what you are talking about.

Changes in this PR:

- Rename all the roles using underscores instead of dashes.

- Rename the playbooks as well.

- Change the directory structure according to what `ansible-galaxy
collection init pbench.agent' gives.

- Add galaxy info to all the meta/main.yml files (make sure
indentation is correct - DAMHIKT).

- Add README.md files to all the roles.

- Add galaxy.yml file to the collection's top dir.

- Delete remote_user from all the playbooks.

There is a simple Makefile that builds a tarball, runs the sanity tests and uploads the tarball to Galaxy. The sanity test ("ansible-test sanity") is absolutely indispensable: Galaxy is very picky about what it will allow - in particular indentation errors are
fatal and cause cryptic error messages and trailing empty lines are also fatal but are diagnosed more precisely. This imposes a "yamllint" dependency, but on the plus side I often invoked that by hand to avoid the "bad telephone line" effect of having Galaxy (or even ansible-test) tell me what's wrong.

dbutenhof reviewed (dismissed):
Nice documentation, along with the required Galaxy tagging. I was abstractly wondering if there was a way to use ansible-galaxy to validate the content aside from just uploading it; and a search sent me down the whole ansible-test rabbit hole, which is a pretty big lift from where we are. Still, using the existing framework to formally test our ansible roles probably isn't a bad idea in the long run. ;-)

riya-17 reviewed (approved):
lgtm

ndokos commented:
It's now ready for review and merging.

There is much left to be done here particularly with testing as @dbutenhof  points out. The rudimentaty testing that *is* done (`ansible-test sanity`) is absolutely indispensable though: I have never seen worse error reporting than what happens if you try to upload a tar ball on the web site: mysterious errors that tell you nothing about what went wrong. Doing `ansible-test sanity` locally is not great either, but at least it gives you something to work with. Doing `yamllint` on a specific file is better (but again far from ideal).

portante commented on agent/ansible/Makefile:
```diff
@@ -0,0 +1,27 @@
+# Collection info
+namespace = pbench
+collection = agent
+version = 1.0.2
```
Should this track the pbench version?

portante replied:
So if that is the case, any way this `Makefile` can derive the version from the `.yml` file to avoid having to keep the version number in two places?

ndokos replied:
Added a yamlconf script to get things out of a yaml file and invoked that from the Makefile.

portante commented on agent/ansible/README:
```diff
@@ -22,5 +22,77 @@ file for guidance.
+  become: yes
+  become_user: root
+
+  # The default value ('production') can be overriddent by cenv, a host-specific
```
s/overriddent/overridden/?

ndokos replied:
Will fix.

portante commented on agent/ansible/README:
```diff
@@ -22,5 +22,77 @@ file for guidance.
+
+Prereqs:
+
+- an account on galaxy.ansible.com - I use my Github account.
```
s/I/typically, one can use a GitHub account for a Galaxy account./

ndokos replied:
Will fix.

portante commented on agent/ansible/pbench/agent/README.md:
````diff
@@ -0,0 +1,94 @@
+
+and an example inventory file that can be edited to suit your circumstances:
+
+```
````
Could we keep an example file in the git repo and remove it from this `README.md` file?

ndokos replied:
See below.

portante commented on agent/ansible/pbench/agent/README.md:
```diff
@@ -0,0 +1,94 @@
+
+Some of these are defaulted for convenience, but pbench_repo_url_prefix and pbench_key_url are specific to an environment and need to be specified locally (most likely in the inventory file).
+
+Here's an example playbook:
```
Could we keep an example playbook in the git repo and remove it from this `README.md`?

ndokos replied:
The thing is that people who download this stuff from galaxy only get this doc file, not the example inventory or example playbook from the git repo. Hence the need to duplicate it in the doc.

portante replied:
Seems like a URL would remove the need to duplicate the playbook in the doc.

ndokos replied:
Done

portante commented on agent/ansible/pbench/agent/galaxy.yml:
```diff
@@ -0,0 +1,57 @@
+name: agent
+
+# The version of the collection. Must be compatible with semantic versioning
+version: 1.0.2
```
Seems like the Makefile should derive the version number it uses from this file?

portante commented on agent/ansible/pbench/agent/galaxy.yml:
```diff
@@ -0,0 +1,57 @@
+repository: http://github.com/distributed-system-analysis/pbench
+
+# The URL to any online docs
+documentation: http://github.com/distributed-system-analysis/pbench
```
Perhaps refer to the gh-pages URL?

ndokos replied:
That's https://distributed-system-analysis.github.io/pbench/doc/  - correct?

portante commented on agent/ansible/pbench/agent/galaxy.yml:
```diff
@@ -0,0 +1,57 @@
+documentation: http://github.com/distributed-system-analysis/pbench
+
+# The URL to the homepage of the collection/project
+homepage: http://github.com/distributed-system-analysis/pbench
```
Since this is for the agent, perhaps we should reference the `agent` subtree of the repository?

portante commented on agent/ansible/pbench/agent/galaxy.yml:
```diff
@@ -0,0 +1,57 @@
+homepage: http://github.com/distributed-system-analysis/pbench
+
+# The URL to the collection issue tracker
+issues: http://github.com/distributed-system-analysis/pbench
```
Perhaps consider `https://github.com/distributed-system-analysis/pbench/issues`?

portante commented on agent/ansible/pbench/agent/roles/pbench_agent_config/README.md:
```diff
@@ -0,0 +1,61 @@
+Example Playbook
+----------------
+
+    - name: install pbench-agent config and key files
```
Seems like we should reference an example playbook that is in the git repo?

ndokos replied:
See above

ndokos replied:
Done

portante commented on agent/ansible/pbench/agent/roles/pbench_agent_files_install/README.md:
```diff
@@ -0,0 +1,53 @@
+parameters to the URL of the containing directory and the list of
+files to copy resp. and calls this role as a subroutine. This role
+copies the files to local storage. We do it that way so that we can
+copy the files to systems outside the firewall that do not have access
```
s/the firewall/a firewall/
s/that dot not have access to the (internal) URLs/that do not have access to the provided URLs/

ndokos replied:
Will fix.

portante commented on agent/ansible/pbench/agent/roles/pbench_agent_files_install/README.md:
```diff
@@ -0,0 +1,53 @@
+Dependencies
+------------
+
+Example Playbook
```
Can we reference a playbook in the git repo?

ndokos replied:
See above.

ndokos replied:
Done

portante commented on agent/ansible/pbench/agent/roles/pbench_agent_install/README.md:
```diff
@@ -0,0 +1,43 @@
+
+- pbench_clean_yum_cache
+
+Example Playbook
```
Can we reference an example playbook in the git repo?

ndokos replied:
See above.

ndokos replied:
Done

portante commented on agent/ansible/pbench/agent/roles/pbench_repo_install/README.md:
```diff
@@ -0,0 +1,40 @@
+Dependencies
+------------
+
+Example Playbook
```
Since we are repeating the example playbook so many times, and it has an error in it, it seems like we should put this example playbook into the git repo itself.

ndokos replied:
See above. Fixing it in five places is suboptimal of course, but having people clone the pbench repo just to get one lousy playbook seems excessive. We could link to the file in the repo. Would that be acceptable?

portante replied:
A URL to the example playbook in the repo would work fine, especially if we give them a raw link to download the file directly.

ndokos replied:
Done.

portante reviewed (changes requested):
This is great!  Seems like we just need a bit of polish to apply the DRY principle.

ndokos commented:
Left the review comments fixes in a separate commit, but I'll squash before merging.

dbutenhof reviewed (dismissed):
Cute little yaml parser in there.

portante commented on agent/ansible/README:
```diff
@@ -41,32 +41,20 @@ The roles will be installed under
 
-You will still a playbook to install the pbench agent - here's the
-standard pbench_agent_install.yml from the pbench repo:
+You will still a playbook to install the pbench agent. You can use the
```
s/You will still a playbook/You will still need a playbook/

portante commented on agent/ansible/pbench/agent/roles/pbench_agent_files_install/README.md:
```diff
@@ -6,8 +6,8 @@ file and an ssh key. pbench_agent_config sets the "source" and "files"
 copies the files to local storage. We do it that way so that we can
-copy the files to systems outside the firewall that do not have access
-to the (internal) URLs.
+copy the files to systems outside afirewall that do not have access
```
s/afirewall/a firewall/

portante reviewed (changes requested):
Just two small fixes ...

ndokos commented:
Fixed the typos, added a missing quote in the Makefile, published a 1.0.3 version with the current state of things, and squashed the commits down to one.

portante reviewed (dismissed):
Nice work, this is fantastic!

dbutenhof commented on agent/ansible/README:
```diff
@@ -22,5 +22,65 @@ file for guidance.
+You will still need a playbook to install the pbench agent. You can use the
+standard pbench_agent_install.yml from the pbench repo - here's a link:
+
+    https://github.com/distributed-system-analysis/pbench/blob/master/agent/ansible/pbench-agent-install.yml
```
Looks like you missed the dash-to-underscore (dash_to_underscore?) change here, and at 57.

ndokos replied:
Old habits die hard...

dbutenhof reviewed (approved):
Minor...

portante reviewed (changes requested):
Needs a rebase before we can merge.
